addon/background.js

```diff
@@ -64,11 +65,12 @@ browser.browserAction.onClicked.addListener(async () => {
   await openUrl(url);
 });
 
-async function openUrl(url) {
+async function openUrl(url, inWindowId = null) {
   // FIXME: should send something in an event about whether the desktop has already been set
-  const windowInfo = await browser.windows.getCurrent();
+  const windowInfo = {id: inWindowId} || await browser.windows.getCurrent();
```

Comment:
This doesn't work. {} || "foo" always evaluates to {}. Perhaps `inWindowId !== null ? { id: inWindowId } : await browser.windows.getCurrent()`?